hub/core/storage/s3/s3.py

```diff
@@ -0,0 +1,52 @@
+from typing import Optional
+from hub.core.storage.provider import Provider
+from hub.core.storage.s3.s3_mapper import S3Mapper
+
+
+class S3Provider(Provider):
+    """
+    Provider class for using S3 storage.
+    """
+
+    def __init__(
+        self,
+        root: str,
+        aws_access_key_id: Optional[str] = None,
+        aws_secret_access_key: Optional[str] = None,
+        aws_session_token: Optional[str] = None,
+        endpoint_url: Optional[str] = None,
+        aws_region: Optional[str] = None,
+        max_pool_connections: Optional[int] = 25,
+    ):
+        """
+        Initializes the S3Provider
+
+        Example:
+            s3_provider = S3Provider("snark-test/benchmarks")
+
+        Args:
+            root (str): the root of the provider.
+            aws_access_key_id (optional, str): Specifies the AWS access key used as part of the credentials to authenticate the user.
+            aws_secret_access_key (optional, str): Specifies the AWS secret key used as part of the credentials to authenticate the user.
+            aws_session_token (optional, str): Specifies an AWS session token used as part of the credentials to authenticate the user.
+            endpoint_url (optional, str): Specify the endpoint url
```

Comment:
missing a ".", also `endpoint_url` is this "s3://bucket"? if it is, this should be noted here.

Reply:
not really, this is for cases in which you use something with an s3 like interface, like minio or wasabi. I haven't used any new nomenclature here, people familiar with it should get it. I'll improve the explanation though